hax-lib/src/lib.rs

```diff
@@ -0,0 +1,104 @@
+//! Hax-specific helpers for Rust programs. Those helpers are usually
+//! no-ops when compiled normally but meaningful when compiled under
+//! hax.
+
+#[cfg(hax)]
+#[macro_export]
+macro_rules! proxy_macro_if_not_hax {
+    ($macro:path, no, $($arg:tt)*) => {
+        ()
+    };
+    ($macro:path, $f:expr, $cond:expr$(, $($arg:tt)*)?) => {
+        $f($cond)
+    };
+}
+#[cfg(not(hax))]
+#[macro_export]
+macro_rules! proxy_macro_if_not_hax {
+    ($macro:path, $f:expr, $($arg:tt)*) => {
+        $macro!($($arg)*)
+    };
+}
+
+#[macro_export]
+/// Proxy to `std::debug_assert!`. Compiled with `hax`, this
+/// disappears.
+macro_rules! debug_assert {
+    ($($arg:tt)*) => {
+        $crate::proxy_macro_if_not_hax!(::std::debug_assert, no, $($arg)*)
+    };
+}
+
+#[macro_export]
+/// Proxy to `std::assert!`. Compiled with `hax`, this is transformed
+/// into a `assert` in the backend.
+macro_rules! assert {
+    ($($arg:tt)*) => {
+        $crate::proxy_macro_if_not_hax!(::std::assert, $crate::assert, $($arg)*)
+    };
+}
+
+#[cfg(hax)]
```

Comment:
Couldn't we inline the functions into the macros then instead of having dead functions here? (See suggestion above)
And if we have them here, we should hide them from the documentation if no one should use them.

Same for assume

Reply:
No, we cannot inline those:
 - if we replace that with a closure, then we cannot translate to the right F* definition;
 - if we inline a `fn` item, then it won't be located in the right place.

I'm going to just hide those from documentation then!